save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/Action.kt

```diff
@@ -0,0 +1,178 @@
+    val (displayMessage, setDisplayMessage) = useState(props.message)
+    val (isError, setError) = useState(false)
+
+    val action = useDeferredRequest {
+        val response = props.sendRequest(props.typeOfOperation)(this)
+        if (response.ok) {
+            props.onActionSuccess(props.typeOfOperation.isClickMode)
+        } else {
+            setDisplayTitle(props.errorTitle)
+            setDisplayMessage(response.unpackMessage())
+            setError(true)
+            windowOpenness.openWindow()
+        }
+    }
+
+    div {
+        button {
+            type = ButtonType.button
+            className = ClassName(props.classes)
+            props.buttonStyleBuilder(this)
+            onClick = {
+                setDisplayTitle(props.title)
+                setDisplayMessage(props.message)
+                windowOpenness.openWindow()
+            }
+        }
+    }
+
+    displayModalWithClick(
+        title = displayTitle,
+        message = displayMessage,
+        clickMessage = props.clickMessage,
+        isOpen = windowOpenness.isOpen(),
+        conditionClickIcon = props.conditionClick && !isError,
+        onCloseButtonPressed = windowOpenness.closeWindowAction(),
+        buttonBuilder = {
+            if (isError) {
+                buttonBuilder("Ok") {
+                    windowOpenness.closeWindow()
+                    setError(false)
+                }
+            } else {
+                props.modalButtons(action, windowOpenness, this)
+            }
+        },
+        changeClickMode = {
+            props.typeOfOperation.changeClickMode(it)
+        }
+    )
+}
+
+/**
+ * Button with modal for something action
+ *
+ * @return noting
+ */
+external interface ActionProps : Props {
+    /**
+     * Type of action
+     */
+    var typeOfOperation: TypeOfAction
+
+    /**
+     * Title of the modal
+     */
+    var title: String
+
+    /**
+     * Error title of the modal
+     */
+    var errorTitle: String
+
+    /**
+     * Message of the modal
+     */
+    var message: String
+
+    /**
+     * Message when clicked
+     */
+    var clickMessage: String
+
+    /**
+     * If the action (request) is successful, this is done
+     */
+    var onActionSuccess: (Boolean) -> Unit
+
+    /**
+     * Button View
+     */
+    var buttonStyleBuilder: (ChildrenBuilder) -> Unit
+
+    /**
+     * Classname for the button
+     */
+    var classes: String
+
+    /**
+     * Modal buttons
+     */
+    @Suppress("TYPE_ALIAS")
+    var modalButtons: (action: () -> Unit, WindowOpenness, ChildrenBuilder) -> Unit
+
+    /**
+     * Condition for click
+     */
+    var conditionClick: Boolean
+
+    /**
+     * Request
+     */
+    var sendRequest: (TypeOfAction) -> DeferredRequestAction<Response>
+}
+
+/**
+ * Type of action
+ */
+enum class TypeOfAction {
+    DELETE_ORGANIZATION,
+    DELETE_PROJECT,
+    RECOVERY_ORGANIZATION,
+    RECOVERY_PROJECT,
+    ;
```

Comment:
Discussed offline - I do not think this is what we need

Reply:
There are some additions to the request, depending on the type of operation, the parameter passed to the backend may be added/changed. Therefore, it is better to pass the type and information.

Reply:
To my mind there is no argument why it's better to pass there enum entity. To my mind it should be done by having nullable callback (onCheckboxChangeCallback or smith like that).

Reply:
To my mind your code is not extendable and this component seems to be very useful as far as you often might ask user to confirm his decision.

Reply:
DELETED